File: 240930/최고의 33위치/best-place-of-33.cpp
#include <iostream>
#include <algorithm>
using namespace std;
#define MAX_N 20

int n, ans;
bool grid[MAX_N][MAX_N];

bool InRange(int x, int y) {
    return 0<=x && x<n && 0<=y && y<n;
}

int FoundCoin(int x, int y) {
    int cnt = 0;
    for(int i=x; i<x+3; i++) {
        for(int j=y; j<y+3; j++) {
            if(!InRange(i, j)) return 0;

            if(grid[i][j]) cnt++;
        }
    }

    return cnt;
}


int main() {
    // 여기에 코드를 작성해주세요.
    cin >> n;
    for(int i=0; i<n; i++) {
        for(int j=0; j<n; j++) {
            cin >> grid[i][j];
        }
    }

    for(int i=0; i<n; i++) {
        for(int j=0; j<n; j++) {
            ans = max(ans, FoundCoin(i, j));
        }
    }

    cout << ans;

    return 0;
}

File: 250114/최고의 33위치/best-place-of-33.py
n = int(input())

grid = [
    list(map(int, input().split()))
    for _ in range(n)
]

ans = 0

def cntCoin(x, y):
    cnt = 0

    for i in range(3):
        for j in range(3):
            if i+x >= n or j+y >= n:
                continue
            
            cnt += grid[x+i][y+j]
    
    return cnt

for i in range(n):
    for j in range(n):
        ans = max(ans, cntCoin(i, j))

print(ans)